File: admin/src/main/java/com/imstargg/admin/domain/BattleMap.java
package com.imstargg.admin.domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.imstargg.storage.db.core.MessageCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.BattleEventCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.BattleMapCollectionEntity;
import com.imstargg.storage.db.core.brawlstars.BrawlStarsImageCollectionEntity;
import jakarta.annotation.Nullable;

import java.util.List;

public record BattleMap(
        BattleMapCollectionEntity entity,
        List<MessageCollectionEntity> names,
        @Nullable BrawlStarsImageCollectionEntity image,
        @JsonIgnoreProperties("map")
        List<BattleEventCollectionEntity> events
) {
}

File: admin/imstargg-admin-frontend/src/model/BattleMap.ts
import BattleMapEntity from "@/model/entity/BattleMapEntity";
import MessageEntity from "@/model/entity/MessageEntity";
import BrawlStarsImageEntity from "@/model/entity/BrawlStarsImageEntity";
import BattleEventEntity from "@/model/entity/BattleEventEntity";

export default interface BattleMap {
  entity: BattleMapEntity;
  names: MessageEntity[];
  image: BrawlStarsImageEntity | null;
  events: BattleEventEntity[];
}